src/Lecture1.hs

```diff
@@ -104,7 +119,14 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str = go (words str) 0
+    where
+        go :: [String] -> Int -> Int
+        go nums result =
+            if null nums
+            then result
+            else go (tail nums) result + read (head nums) :: Int
```

Comment:
There's already a standard function `sum` to find a sum of all elements in the list.
Additionally, you can use the `map` function to parse every string in the list. Using standard function, you'll be able to implement this function as a one-liner 🙂

Reply:
that's an awesome suggestion! thank you